### ISSUE TITLE
`selector-no-redundant-nesting-selector` incorrectly errors within a `:where()` selector

### ISSUE THREAD

**Issue by juzerzarif** [stylelint-scss/stylelint-scss#1000]
Hi, I have a selector like
```scss
.my-input {
  border-color: red;

  :where(&)::placeholder {
    color: gray;    
  }
}
```
The `selector-no-redundant-nesting-selector` rule complains about `Unnecessary nesting selector (&)`. I could use `&::placeholder` as the selector but that has more specificity than `:where(&)::placeholder` ((0,2,0) vs (0,1,0)) so the `:where` is necessary here.

**Comment by kristerkari** [comment]
Thanks! I guess that the rule is just not made to handle such selector, so this can be marked as a bug 😄
